### DISCUSSION TITLE
'Network Out Total' cut in half from MySql.Data.MySqlClient -> MySqlConnector. How?

### DISCUSSION THREAD

**Discussion by adr-x** [mysql-net/MySqlConnector#1532] [Q&A]
Changing from MySql.Data.MySqlClient to MySqlConnector cut the network out total on the box with our MySql server in half. 
Does anyone have some intuition that they could share for how this could be possible?

![image](https://github.com/user-attachments/assets/122ec3c8-c38d-4181-bbc1-5175191d5ff7)

**Comment by bgrainger** [comment]
MySql.Data performs a lot of unnecessary (IMO) reads from the MySQL Server when you open a connection, e.g., `LoadServerPropertiesAsync` https://github.com/mysql/mysql-connector-net/blob/ac184e8a3626790551d381732b08680922882dd9/MySQL.Data/src/Driver.cs#L298 and `LoadCharacterSetsAsync` https://github.com/mysql/mysql-connector-net/blob/ac184e8a3626790551d381732b08680922882dd9/MySQL.Data/src/Driver.cs#L344C24-L344C46.

MySqlConnector has been highly optimized for performance (see https://mysqlconnector.net/#performance) and tries to reduce network I/O to a minimum.
